Code/GraphMol/DetermineBonds/DetermineBonds.cpp

```diff
@@ -66,6 +74,332 @@ void determineConnectivity(RWMol &mol, bool useHueckel, int charge, double covFa
+    int molCharge = 0;
+    for (int i = 0; i < mol.getNumAtoms(); i++) {
+        auto atom = mol.getAtomWithIdx(i);
+        int atomCharge = getAtomicCharge(atom->getAtomicNum(), valency[i]);
+        molCharge += atomCharge;
+        if (atom->getAtomicNum() == 6) {
+            if (atom->getDegree() == 2 && valency[i] == 2) {
+                molCharge += 1;
+                atomCharge = 2;
+            } else if (atom->getDegree() == 3 && (molCharge + 1 < charge)) {
+                molCharge += 2;
+                atomCharge = 1;
+            }
+        }
+    }
+    return molCharge == charge;
+}
+
+void setAtomicCharges(RWMol &mol, std::vector<int> &valency, int charge) {
+    int molCharge = 0;
+    for (int i = 0; i < mol.getNumAtoms(); i++) {
+        auto atom = mol.getAtomWithIdx(i);
+        int atomCharge = getAtomicCharge(atom->getAtomicNum(), valency[i]);
+        molCharge += atomCharge;
+        if (atom->getAtomicNum() == 6) {
+            if (atom->getDegree() == 2 && valency[i] == 2) {
+                molCharge += 1;
+                atomCharge = 0;
+            } else if (atom->getDegree() == 3 && (molCharge + 1 < charge)) {
+                molCharge += 2;
+                atomCharge = 1;
+            }
+        }
+        if (atomCharge != 0) {
+            atom->setFormalCharge(atomCharge);
+        }
+    }
+}
+
+void setAtomicRadicals(RWMol &mol, std::vector<int> &valency, int charge) {
+    for (int i = 0; i < mol.getNumAtoms(); i++) {
+        auto atom = mol.getAtomWithIdx(i);
+        int atomCharge = getAtomicCharge(atom->getAtomicNum(), valency[i]);
+        if (atomCharge != 0) {
+            atom->setNumRadicalElectrons(std::abs(charge));
+        }
+    }
+}
+
+bool checkSaturation(std::vector<int> &order, std::vector<int> &valency) {
+    std::vector<int> unsat;
+    getUnsaturated(order, valency, unsat);
+    if (unsat.size() == 0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void setAtomMap(RWMol &mol) {
+    for (int i = 0; i < mol.getNumAtoms(); i++) {
+        auto atom = mol.getAtomWithIdx(i);
+        atom->setAtomMapNum(i + 1);
+    }
+}
+
+void setChirality(RWMol &mol) {
+    MolOps::sanitizeMol(mol);
+    MolOps::setDoubleBondNeighborDirections(mol, &mol.getConformer());
+    MolOps::assignStereochemistryFrom3D(mol);
+    MolOps::assignChiralTypesFrom3D(mol);
+}
+
+void addBondOrdering(RWMol &mol, std::vector<std::vector<int>> &ordMat,
+                     std::vector<int> &valency, bool allowChargedFragments,
+                     bool embedChiral, bool useAtomMap, int charge) {
+    for (int i = 0; i < mol.getNumAtoms(); i++) {
+        for (int j = i + 1; j < mol.getNumAtoms(); j++) {
+            if (ordMat[i][j] == 0 || ordMat[i][j] == 1) {
+                continue;
+            } else if (ordMat[i][j] == 2) {
+                mol.removeBond(i, j);
+                mol.addBond(i, j, Bond::BondType::DOUBLE);
```

Comment:
Seems like this would be equivalent and way more efficient:
```suggestion
                mol.getBondBetweenAtoms(i,j)->setBondType(Bond::BondType::DOUBLE);                ```